src/node/heart.ts

```diff
@@ -20,20 +23,28 @@ export class Heart {
    * timeout and start or reset a timer that keeps running as long as there is
    * activity. Failures are logged as warnings.
    */
-  public beat(): void {
+  public async beat(): Promise<void> {
     if (this.alive()) {
       return
     }
 
     logger.trace("heartbeat")
-    fs.writeFile(this.heartbeatPath, "").catch((error) => {
+    try {
+      await fs.writeFile(this.heartbeatPath, "")
```

Comment:
Thinking about some more, we might want to return this promise at the very end to avoid concurrency issues.

For example if `beat` was called multiple times subsequent calls will think it is not alive because the original call has not yet updated `lastHeartbeat') and they will all try writing to the file and start their own timers

Reply:
Ah, good call!